File: Programmers/Level1/220819/javascript/CaesarCipher.js
// eslint-disable-next-line import/prefer-default-export
export function solution(s, n) {
  const lowerCase = 'abcdefghijklmnopqrstuvwxyz';
  const upperCase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';

  const answer = {
    ciper: '',
  };

  for (let i = 0; i < s.length; i += 1) {
    if (s.charAt(i) === ' ') {
      answer.ciper += ' ';
      continue;
    }

    const sentence = lowerCase.includes(s.charAt(i)) ? lowerCase : upperCase;

    const char = {
      alphabet: sentence.indexOf(s.charAt(i)),
    };

    char.alphabet += n;

    if (char.alphabet > 25) {
      char.alphabet -= 26;
    }

    answer.ciper += sentence.charAt(char.alphabet);
  }

  return answer.ciper;
}

File: Programmers/Level1/220819/java/CaesarCipher.java
public class CaesarCipher {
  public String solution(String s, int n) {
    String answer = "";

    String lower = "abcdefghijklmnopqrstuvwxyz";
    String upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    for(int i = 0; i < s.length(); i += 1) {
      if(s.charAt(i) == ' ') {
        answer += ' ';
        continue;
      }
      if(Character.isUpperCase(s.charAt(i))) {
        answer += upper.charAt((upper.indexOf(s.charAt(i)) + n) % 26);
      }
      if(Character.isLowerCase(s.charAt(i))) {
        answer += lower.charAt((lower.indexOf(s.charAt(i)) + n) % 26);
      }
    }

    return answer;
  }
}
